code_samples/sql-data-types.js

```diff
@@ -0,0 +1,366 @@
+    const rowMetadata = await result.getRowMetadata();
+    console.log(SqlColumnType[rowMetadata.getColumnByIndex(rowMetadata.findColumn('this')).type]); // VARCHAR
+
+    for await (const row of result) {
+        console.log(row);
+    }
+};
+
+const booleanExample = async (client) => {
+    console.log('----------BOOLEAN Example----------');
+    const someMap = await client.getMap('booleanMap');
+
+    for (let key = 0; key < 10; key++) {
+        await someMap.set(key, key % 2 === 0);
+    }
+
+    const result = client.getSqlService().execute('SELECT * FROM booleanMap WHERE this = ?', [true]);
+    const rowMetadata = await result.getRowMetadata();
+    console.log(SqlColumnType[rowMetadata.getColumnByIndex(rowMetadata.findColumn('this')).type]); // BOOLEAN
+
+    for await (const row of result) {
+        console.log(row);
+    }
+};
+
+/*
+    Note for TINYINT, SMALLINT, INTEGER AND BIGINT
+
+    Since Node.js client sends all numbers as double by default, giving a number as parameter will not work for these
+    types. Instead, you can use `long` objects or you can use explicit casting which converts doubles to integers. The
+    casting behaviour is Java's casting behaviour. So, e.g, 3.1 becomes 3.
+*/
+const integersExample = async (client) => {
+    console.log('----------TINYINT, SMALLINT, INTEGER AND BIGINT Example----------');
+    const someMap = await client.getMap('bigintMap');
+
+    for (let key = 0; key < 10; key++) {
+        await someMap.set(key, long.fromNumber(key * 2));
+    }
+
+    const result = client.getSqlService().execute(
+        'SELECT * FROM bigintMap WHERE this > ? AND this < ?',
+        [long.fromNumber(10), long.fromNumber(18)]
+    );
+    const rowMetadata = await result.getRowMetadata();
+    console.log(SqlColumnType[rowMetadata.getColumnByIndex(rowMetadata.findColumn('this')).type]); // BIGINT
+
+    for await (const row of result) {
+        console.log(row);
+    }
+
+    // Casting example
+    const result2 = client.getSqlService().execute(
+        'SELECT * FROM bigintMap WHERE this > CAST(? AS BIGINT) AND this < CAST(? AS BIGINT)',
+        [10, 18]
+    );
+
+    for await (const row of result2) {
+        console.log(row);
+    }
+};
+
+// Arbitrary precision decimal
+const decimalExample = async (client) => {
+    console.log('----------DECIMAL Example----------');
+    // Pretend that there are decimal entries in the map added by another client. Then the query would be as follows.
+    const result = client.getSqlService().execute(
+        // We need these casts since we are sending strings.
+        'SELECT * FROM decimalMap WHERE this > CAST(? AS DECIMAL) AND this < CAST(? AS DECIMAL)',
+        ['-0.00000000000000000000000000000001', '1.0000000000000231213123123125465462513214653123']
+    );
+
+    try {
+        const rowMetadata = await result.getRowMetadata();
+        console.log(SqlColumnType[rowMetadata.getColumnByIndex(rowMetadata.findColumn('this')).type]); // DOUBLE
```

Comment:
It is impossible to add decimal using nodejs client, so I am not sure what to write there; because writing DECIMAL would be wrong as it wont print DECIMAL